apps/api/src/app/partner-integrations/usecases/update-vercel-configuration/update-vercel-configuration.usecase.ts

```diff
@@ -0,0 +1,261 @@
+        configurationDetails.accessToken,
+        configurationDetails.teamId,
+        projectIds
+      );
+
+      const mappedProjectData = this.mapProjectKeys(envKeys, command.data, { updateProjectDetails, addProjectIds });
+
+      for (const key of Object.keys(mappedProjectData)) {
+        await this.setEnvironments({
+          clientKey: mappedProjectData[key].clientKey,
+          privateKey: mappedProjectData[key].privateKey,
+          projectIds: mappedProjectData[key].addProjectIds,
+          teamId: configurationDetails.teamId,
+          token: configurationDetails.accessToken,
+        });
+        await this.updateEnvironments({
+          clientKey: mappedProjectData[key].clientKey,
+          privateKey: mappedProjectData[key].privateKey,
+          projectDetails: mappedProjectData[key].updateProjectDetails,
+          teamId: configurationDetails.teamId,
+          token: configurationDetails.accessToken,
+        });
+      }
+
+      await this.saveProjectIds(command);
+
+      return {
+        success: true,
+      };
+    } catch (error) {
+      throw new ApiException(error.message);
+    }
+  }
+
+  private async getEnvKeys(organizationIds: string[]): Promise<EnvironmentEntity[]> {
+    return await this.environmentRepository.find(
+      {
+        _organizationId: organizationIds,
+      },
+      'apiKeys identifier _organizationId'
+    );
+  }
+
+  private mapProjectKeys(
+    envData: EnvironmentEntity[],
+    projectData: Record<string, string[]>,
+    projectDetails: MappedProjectData
+  ) {
+    const { addProjectIds, updateProjectDetails } = projectDetails;
+    const mappedData = envData.reduce<Record<string, MapProjectkeys>>((acc, curr) => {
+      const projectIds = projectData[curr._organizationId];
+      const newData = {
+        privateKey: curr.apiKeys[0].key,
+        clientKey: curr.identifier,
+        updateProjectDetails: updateProjectDetails.filter((detail) => projectIds.includes(detail.projectId)),
+        addProjectIds: projectIds.filter((id) => addProjectIds.includes(id)),
+      };
+      acc[curr._organizationId] = newData;
+
+      return acc;
+    }, {});
+
+    return mappedData;
+  }
+
+  private async setEnvironments({ clientKey, projectIds, privateKey, teamId, token }: ISetEnvironment): Promise<void> {
+    const projectApiUrl = `${process.env.VERCEL_BASE_URL}/v9/projects`;
+    const target = ['production', 'preview', 'development'];
+    const type = 'encrypted';
+
+    const apiKeys = [
+      {
+        target,
+        type,
+        value: clientKey,
+        key: 'NOVU_CLIENT_APP_ID',
+      },
+      {
+        target,
+        type,
+        value: privateKey,
+        key: 'NOVU_API_SECRET',
+      },
+    ];
+
+    await Promise.all(
+      projectIds.map((projectId) =>
+        lastValueFrom(
+          this.httpService.post(`${projectApiUrl}/${projectId}/env${teamId ? `?teamId=${teamId}` : ''}`, apiKeys, {
+            headers: {
```

Comment:
this will be used the `addprojectsId` i.e the new projects added during the update phase